llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -604,7 +606,8 @@ SIRegisterInfo::getMaxNumVectorRegs(const MachineFunction &MF) const {
 
     if (MinNumAGPRs == DefaultNumAGPR.first) {
       // Default to splitting half the registers if AGPRs are required.
-      MinNumAGPRs = MaxNumAGPRs = MaxVectorRegs / 2;
+      MinNumAGPRs = MaxNumAGPRs =
+          (MFI->mayNeedAGPRs() || MinWavesPerEU == 1) ? MaxVectorRegs / 2 : 0;
```

Comment:
This is used to compute mayNeedAGPRs, it cannot be used here. I just moved this function to no longer depend on the MF